pkg/k8s/scanner/scanner.go

```diff
@@ -90,7 +90,19 @@ func (s *Scanner) Scan(ctx context.Context, artifactsData []*artifacts.Artifact)
 	onItem := func(ctx context.Context, artifact *artifacts.Artifact) (scanResult, error) {
 		scanResults := scanResult{}
 		if s.opts.Scanners.AnyEnabled(types.VulnerabilityScanner, types.SecretScanner) {
-			vulns, err := s.scanVulns(ctx, artifact)
+			opts := s.opts
```

Comment:
Copying the Option struct is not enough since `s.opts.Credentials` and `opts` refers to the same pointer of `Credentials`.
https://go.dev/play/p/Z0Zx4W4pz_L

```suggestion			
			// Deep-copy s.opts.Credentials
			opts := s.opts
			opts.Credentials := make([]string, len(s.opts.Credentials))
			copy(opts.Credentials, s.opts.Credentials)
```